test/e2e/plain_provisioner_test.go

```diff
@@ -351,115 +351,100 @@ var _ = Describe("plain provisioner bundle", func() {
 
 	When("Bundles are backed by a git repository", func() {
 		var (
-			bundles []*rukpakv1alpha1.Bundle
-			ctx     context.Context
+			ctx context.Context
 		)
 		BeforeEach(func() {
 			ctx = context.Background()
+		})
 
-			By("creating the git based Bundles")
-			bundles = []*rukpakv1alpha1.Bundle{
-				{
-					ObjectMeta: metav1.ObjectMeta{
-						GenerateName: "combo-git-commit",
-					},
-					Spec: rukpakv1alpha1.BundleSpec{
-						ProvisionerClassName: plainProvisionerID,
-						Source: rukpakv1alpha1.BundleSource{
-							Type: rukpakv1alpha1.SourceTypeGit,
-							Git: &rukpakv1alpha1.GitSource{
-								Repository: "https://github.com/exdx/combo-bundle",
-								Ref: rukpakv1alpha1.GitRef{
-									Commit: "9e3ab7f1a36302ef512294d5c9f2e9b9566b811e",
-								},
-							},
+		DescribeTable("Can create and unpack a git bundle successfully",
+			func(bundle *rukpakv1alpha1.Bundle) {
+				err := c.Create(ctx, bundle)
+				Expect(err).To(BeNil())
+
+				Eventually(func() bool {
+					if err := c.Get(ctx, client.ObjectKeyFromObject(bundle), bundle); err != nil {
+						return false
+					}
+					return bundle.Status.Phase == rukpakv1alpha1.PhaseUnpacked
+				}).Should(BeTrue())
+
+				Eventually(func() bool {
+					if err := c.Get(ctx, client.ObjectKeyFromObject(bundle), bundle); err != nil {
+						return false
+					}
+					if bundle.Status.Info == nil {
+						return false
+					}
+					/*
+						manifests
+						├── 00_namespace.yaml
+						├── 01_cluster_role.yaml
+						├── 01_service_account.yaml
+						├── 02_deployment.yaml
+						├── 03_cluster_role_binding.yaml
+						├── combo.io_combinations.yaml
+						└── combo.io_templates.yaml
+					*/
+					return len(bundle.Status.Info.Objects) == 7
+				}).Should(BeTrue())
+
+				// delete bundle
+				Eventually(func() bool {
+					err := c.Delete(ctx, bundle)
+					return apierrors.IsNotFound(err)
+				}).Should(BeTrue())
```

Comment:
It seems like we risk this Bundle being stranded if the Eventually statement above fails.

Reply:
That's a good point. I can explore trying to delete the bundle in an AfterEach, the trick is how to get the relevant bundle to delete. It's possible to just do a List and delete whatever comes back? Or label the bundle when creating it and then using a label selector to query for just those bundles (that sounds good since it's less destructive and doesn't impact bundles outside that particular e2e scope).

Reply:
Yeah I think my preference is towards labeling them and firing off a list query. I'm not sure whether this introduces non-determinism into our testing suite.

Reply:
Yes, I'm not super thrilled about that, but I'm not sure there's another way given the limitations around `AfterEach` getting data from a `DescribeTable` 🤷